Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -291,10 +299,17 @@ void HTMLAttachmentElement::updateProgress(const AtomString& progress)
     bool validProgress = false;
     float value = progress.toFloat(&validProgress);
     if (validProgress && std::isfinite(value)) {
+        if (value == 0.0) {
+            m_placeholderElement->removeAttribute(styleAttr);
+            m_progressElement->setAttributeWithoutSynchronization(styleAttr, "display: none;"_s);
```

Comment:
Vaguely surprised we're setting `style` like this instead of `setInlineStyleProperty` like PDFPlugin&friends; isn't it hard to keep track/avoid overwriting things you set from elsewhere?

Reply:
I'll look into `setInlineStyleProperty`, nifty!

Reply:
`setInlineStyleProperty` would be good. I think you'll have to keep the `--progress:` below using the attribute setter though.

Reply:
`setInlineStyleCustomProperty` seems to work for `--progress`.

Reply:
Nice!